llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1932,8 +2000,15 @@ void AsmPrinter::emitFunctionBody() {
 
   // Emit section containing BB address offsets and their metadata, when
   // BB labels are requested for this function. Skip empty functions.
-  if (MF->hasBBLabels() && HasAnyRealCode)
+  bool HasBBLabels = MF->hasBBLabels();
+  if (HasBBLabels && HasAnyRealCode)
     emitBBAddrMapSection(*MF);
+  else if (!HasBBLabels && HasAnyRealCode &&
+           PgoAnalysisMapFeatures.getBits() != 0)
+    MF->getContext().reportWarning(
+        SMLoc(), "pgo-analysis-map is enabled but the following machine "
+                 "function was does not have labels: " +
```

Comment:
typo. (was done).

Also, when saying "machine function" users probably expect MIR to follow. So Maybe change to:
"pgo-analysis map is enabled the function " << MF->getName() << " does not have basic block labels".